crates/rover-client/src/operations/readme/publish/runner.rs

```diff
@@ -0,0 +1,118 @@
+use super::types::*;
+use crate::blocking::StudioClient;
+use crate::operations::readme::publish::ReadmePublishInput;
+use crate::shared::GraphRef;
+use crate::RoverClientError;
+use graphql_client::*;
+
+type Timestamp = String;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/operations/readme/publish/publish_mutation.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+
+pub struct ReadmePublishMutation;
+
+pub fn run(
+    input: ReadmePublishInput,
+    client: &StudioClient,
+) -> Result<ReadmePublishResponse, RoverClientError> {
+    let graph_ref = input.graph_ref.clone();
+    let response = client.post::<ReadmePublishMutation>(input.into())?;
+    build_response(response, graph_ref)
+}
+
+fn build_response(
+    data: readme_publish_mutation::ResponseData,
+    graph_ref: GraphRef,
+) -> Result<ReadmePublishResponse, RoverClientError> {
+    let readme = data
+        .graph
+        .ok_or(RoverClientError::GraphNotFound {
+            graph_ref: graph_ref.clone(),
+        })?
+        .variant
+        .ok_or(RoverClientError::GraphNotFound { graph_ref })?
+        .update_variant_readme
+        .ok_or(RoverClientError::AdhocError {
+            msg: "error publishing README".to_string(),
+        })?
+        .readme
+        .ok_or(RoverClientError::AdhocError {
+            msg: "error publishing README".to_string(),
```

Comment:
i would love if we could have more specific error messages here - when are these null cases encountered? if they are encountered is it a bug on the studio side? if so, we should use `RoverClientError::MalformedResponse { null_field: "update_variant_readme" }` - otherwise should have something more descriptive here with a path to resolution for the end user